src/project_types/project.rs

```diff
@@ -143,7 +143,21 @@ impl Project {
                 return Ok(new_project);
             }
         }
-        Err(Error::SequenceNotFound(name.to_string()))
+        Err(Error::SequenceNotFound(name.to_owned()))
+    }
+
+    pub fn resection_sequence(&self, name: &str, num_sections: u32) -> Result<Project, Error> {
+        let mut new_project = self.clone();
+        for i in 0..new_project.sequences.len() {
+            if new_project.sequences[i].name == name {
+                {
+                    let sequence = &mut new_project.sequences[i];
+                    try!(sequence.resection(num_sections));
+                }
+                return Ok(new_project);
+            }
+        }
```

Comment:
```
match new_project.sequences.iter_mut().find(|seq| seq.name == name) {
    None => Err(Error::SequenceNotFound(name.to_owned())),
    Some(ref mut seq) => {
         seq.resection(num_sections);
         Ok(new_project)
     }
}
```

Possibly a double borrow issue on the Ok line, so you can move that out if you need to.